.github/workflows/build-deploy.yml

```diff
@@ -38,9 +43,40 @@ jobs:
           go-version-file: "go.mod"
           cache: true
 
-      - name: Test
+      - name: Run Unit and Integration Test
         run: make verify test
 
+      - name: Auth to GCP account using WIF for E2E tests
+        if: github.repository == github.event.pull_request.head.repo.full_name
+        uses: "google-github-actions/auth@8254fb75a33b976a221574d287e93919e6a36f70"
+        with:
+          project_id: ${{ env.E2EPROJECTID }}
+          workload_identity_provider: "projects/379706009135/locations/global/workloadIdentityPools/gh-gcp-config-connector-tagging-/providers/gh-gcp-config-connector-tagging-" # "projects/123456789/locations/global/workloadIdentityPools/github/providers/my-repo"
```

Comment:
Should we also add the project-id as an env var? maybe also the pool name? Not sure how modular this needs to be

Reply:
Done